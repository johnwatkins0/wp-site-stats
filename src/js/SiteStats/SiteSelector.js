import React from 'react';

import { AppContext } from '../Context';

const SiteSelector = () => (
	<AppContext.Consumer>
		{ ({ activeSiteId, sites, setActiveSite }) => (
			<label>
				Sites:{' '}
				<select
					className="SiteSelector"
					value={activeSiteId}
					onChange={event => {
						setActiveSite( event.target.value );
					}}
				>

					{sites.map( ({ id, domain, path }) => (
						<option
							key={id}
							value={id}
						>
							{`${domain}${path}`}
						</option>
					) )}
				</select>
			</label>
		)}
	</AppContext.Consumer>
);

export default SiteSelector;
