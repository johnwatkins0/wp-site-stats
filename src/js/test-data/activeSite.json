{
    "site_title": "My Site",
    "site_url": "http://wpmu/my-site",
    "user_count": 1,
    "post_count": {
        "publish": "1",
        "future": 0,
        "draft": 0,
        "pending": 0,
        "private": 0,
        "trash": 0,
        "auto-draft": 0,
        "inherit": 0
    },
    "page_count": {
        "publish": "1",
        "future": 0,
        "draft": 0,
        "pending": 0,
        "private": 0,
        "trash": 0,
        "auto-draft": 0,
        "inherit": 0
    },
    "comment_count": {
        "approved": "1",
        "spam": 0,
        "trash": 0,
        "post-trashed": 0,
        "total_comments": 1,
        "all": 1,
        "moderated": 0
    },
    "latest_post": {
        "id": 1,
        "date": "2018-04-02T14:29:11",
        "date_gmt": "2018-04-02T14:29:11",
        "guid": {
            "rendered": "http://wpmu/my-site/?p=1"
        },
        "modified": "2018-04-02T14:29:11",
        "modified_gmt": "2018-04-02T14:29:11",
        "slug": "hello-world",
        "status": "publish",
        "type": "post",
        "link": "http://wpmu/my-site/2018/04/02/hello-world/",
        "title": {
            "rendered": "Hello world!"
        },
        "content": {
            "rendered": "<p>Welcome to <a href=\"http://wpmu/\">Colby College Sites</a>. This is your first post. Edit or delete it, then start blogging!</p>\n",
            "protected": false
        },
        "excerpt": {
            "rendered": "<p>Welcome to Colby College Sites. This is your first post. Edit or delete it, then start blogging!</p>\n",
            "protected": false
        },
        "author": 1,
        "featured_media": 0,
        "comment_status": "open",
        "ping_status": "open",
        "sticky": false,
        "template": "",
        "format": "standard",
        "meta": [],
        "categories": [
            1
        ],
        "tags": [],
        "revisions": null
    },
    "latest_comment": {
        "id": 1,
        "post": 1,
        "parent": 0,
        "author": 0,
        "author_name": "A WordPress Commenter",
        "author_url": "http://wpmu/",
        "date": "2018-04-02T14:29:11",
        "date_gmt": "2018-04-02T14:29:11",
        "content": {
            "rendered": "<p>Hi, this is a comment.<br />\nTo get started with moderating, editing, and deleting comments, please visit the Comments screen in the dashboard.<br />\nCommenter avatars come from <a href=\"https://gravatar.com\">Gravatar</a>.</p>\n"
        },
        "link": "http://wpmu/my-site/2018/04/02/hello-world/#comment-1",
        "status": "approved",
        "type": "comment",
        "author_avatar_urls": {
            "24": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=24&d=mm&r=g",
            "48": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=48&d=mm&r=g",
            "96": "http://1.gravatar.com/avatar/d7a973c7dab26985da5f961be7b74480?s=96&d=mm&r=g"
        },
        "meta": []
    }
}