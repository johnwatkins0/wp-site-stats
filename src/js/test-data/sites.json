[
    {
        "id": "1",
        "domain": "wpmu",
        "path": "/",
        "registered": "2018-04-02 14:28:19"
    },
    {
        "id": "2",
        "domain": "wpmu",
        "path": "/my-site/",
        "registered": "2018-04-02 14:29:11"
    },
    {
        "id": "3",
        "domain": "wpmu",
        "path": "/another-site/",
        "registered": "2018-04-02 14:29:27"
    },
    {
        "id": "4",
        "domain": "wpmu",
        "path": "/cool-site/",
        "registered": "2018-04-03 17:13:22"
    },
    {
        "id": "5",
        "domain": "wpmu",
        "path": "/another-cool-site/",
        "registered": "2018-04-03 17:13:32"
    },
    {
        "id": "6",
        "domain": "wpmu",
        "path": "/better-site/",
        "registered": "2018-04-03 17:13:44"
    },
    {
        "id": "7",
        "domain": "wpmu",
        "path": "/best-site/",
        "registered": "2018-04-03 17:13:53"
    }
]