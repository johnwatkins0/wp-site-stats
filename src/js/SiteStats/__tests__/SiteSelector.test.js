import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';

import SiteSelector from '../SiteSelector';

import sites from '../../test-data/sites.json';
import activeSite from '../../test-data/activeSite.json';

Enzyme.configure({adapter: new Adapter() });

describe( 'SiteSelector', () => {
	it( 'renders', () => {
		const tree = renderer.create(
			<SiteSelector
				sites={sites}
				activeSiteId={activeSite.id}
				setActiveSite={( id ) => null}
			/>
		).toJSON();
		expect( tree ).toMatchSnapshot();
	});

	it( 'handles a change', () => {
		const setActiveSite = sinon.spy();

        const wrapper = shallow(
            <SiteSelector
				sites={sites}
				activeSiteId={activeSite.id}
				setActiveSite={setActiveSite}
            />
		);

		const mockChangeEvent = { target: { value: '3' } };

		wrapper.find( 'select' ).simulate( 'change', mockChangeEvent );
		expect( setActiveSite.calledOnce ).toEqual( true );
    });

});
