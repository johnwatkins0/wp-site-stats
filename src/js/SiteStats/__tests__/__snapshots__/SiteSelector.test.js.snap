// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteSelector renders 1`] = `
<label>
  Sites:
   
  <select
    className="SiteSelector"
    onChange={[Function]}
    value={undefined}
  >
    <option
      value="1"
    >
      wpmu/
    </option>
    <option
      value="2"
    >
      wpmu/my-site/
    </option>
    <option
      value="3"
    >
      wpmu/another-site/
    </option>
    <option
      value="4"
    >
      wpmu/cool-site/
    </option>
    <option
      value="5"
    >
      wpmu/another-cool-site/
    </option>
    <option
      value="6"
    >
      wpmu/better-site/
    </option>
    <option
      value="7"
    >
      wpmu/best-site/
    </option>
  </select>
</label>
`;
